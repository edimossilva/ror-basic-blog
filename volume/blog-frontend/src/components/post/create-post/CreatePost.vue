<template lang="">
  <div>
    <p>Create Post</p>

    <div>
      <label> title: </label>
      <input type="text" v-model="title"> 
    </div>

    <button @click="createPost">Create</button>
  </div>
</template>

<script>
import potsApi from "../../../services/posts_api";

export default {
  props: {
    blogId: {
      required: true
    }
  },
  data() {
    return {
      title: ""
    };
  },
  methods: {
    createPost() {
      const params = {
        title: this.title,
        blogId: this.blogId
      };
      potsApi
        .createPost({
          params,
          token: this.$store.getters.authToken
        })
        .then(this.redirectToShowPost)
        .catch(this.showErrorMessage);
    },
    redirectToShowPost({ data: { data } }) {
      this.$router.push({
        name: "showPost",
        params: { id: data.id, blogId: data.blog_id }
      });
    },
    showErrorMessage({
      response: {
        data: { error_message }
      }
    }) {
      // debugger; // eslint-disable-line
      this.$modal.show("modal", { message: error_message });
    }
  }
};
</script>

<style scoped></style>