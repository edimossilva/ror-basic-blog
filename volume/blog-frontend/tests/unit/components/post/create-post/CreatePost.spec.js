import { shallowMount } from '@vue/test-utils'
import CreatePost from '@/components/post/create-post/CreatePost'

describe('CreatePost.vue', () => {
  it('is a valid component', () => {
    const wrapper = shallowMount(CreatePost, {
      propsData: { blogId: 1 },
    });

    expect(wrapper.isVueInstance()).toBeTruthy()
    expect(wrapper).toMatchSnapshot()
  })
})
