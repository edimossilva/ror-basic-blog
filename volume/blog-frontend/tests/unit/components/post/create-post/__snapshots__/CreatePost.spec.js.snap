// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreatePost.vue is a valid component 1`] = `
<div>
  <p>Create Post</p>
  <div><label> title: </label> <input type="text"></div> <button>Create</button>
</div>
`;
